src/Paramore.Brighter.MessagingGateway.AWSSQS/AWSMessagingGateway.cs

```diff
@@ -45,16 +45,16 @@ public AWSMessagingGateway(AWSMessagingGatewayConnection awsConnection)
             _awsClientFactory = new AWSClientFactory(awsConnection);
         }
 
-        protected string EnsureTopic(RoutingKey topic, SnsAttributes attributes, TopicFindBy topicFindBy, OnMissingChannel makeTopic)
+        protected async Task<string> EnsureTopic(RoutingKey topic, SnsAttributes attributes, TopicFindBy topicFindBy, OnMissingChannel makeTopic)
```

Comment:
Since this is a v9 PR, you need to be careful about breaking changes. Changing the signature of a non-private method on a non-private class is a breaking change for consumers updating their packages - they'd now have a bunch of tasks flying around in their code they're not going anything with.

Breaking changes should be reserved for major version upgrades, so v10 would be an OK place to make changes like this. That said, it would need to be part of a broader conversation around the conventions when it comes to supporting async methods. I would suggest we keep it as the addition of the async publish methods, and leave the auxiliary methods as they are now.